lib/src/main/java/io/ably/lib/rest/AblyRest.java

```diff
@@ -149,6 +152,51 @@ public void statsAsync(Param[] params, Callback<AsyncPaginatedResult<Stats>> cal
 	}
 
 	/**
+	 * Make a generic HTTP request for a collection of the given type
+	 * @param params query options: see Ably REST API documentation
+	 * for available options
+	 * @return a PaginatedResult of Stats records for the requested params
+	 * @throws AblyException
+	 */
+	public PaginatedResult<JsonElement> paginatedRequest(String method, String path, Param[] params, RequestBody body, Param[] headers) throws AblyException {
```

Comment:
Would it not simply be better for us to have a PaginatedResponse where items is a `JsonStructure`, and therefore just have one `request` method?  I really dislike the idea that a user has to determine what type of response they are getting beforehand.